File: Data Structures & Algorithms/remove-element/submission-0.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int currsize = nums.length;
        int count = currsize;
        int j = 0;
        for(int i =0; i<currsize; i++){
            if(nums[i] == val){
                count -= 1;
            }else{
                nums[j] = nums[i];
                j+=1;
            }
        }
        return count;
    }
}

File: Data Structures & Algorithms/first-missing-positive/submission-0.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int len = nums.size();
        
        for(int i=0; i<len; i++){
            while(nums[i] != i+1 && nums[nums[i] - 1] != nums[i]){
                if(nums[i] <= 0 || nums[i] > len){
                    break;
                }
                int index = nums[i] - 1;
                int temp = nums[index];
                nums[index] = nums[i]; 
                nums[i] = temp;
            }
        }

        for(int i=0; i<nums.size(); i++){
            if(nums[i] != i+1){
                return i+1;
            }
        }
        return len +1;
    }
};